### PR TITLE
Adding support for executing attribute/query calls without assigning to a prior variable

### PR DESCRIPTION
**Pull request by anutosh491**
I realize that we couldn't access attributes or query methods without assigning the expression to a variable beforehand.
So this works 
```
from sympy import pi, Symbol, sin, cos

def main0():
    x: S = Symbol("x")
    a: S = sin(x)
    print(a.diff(x))

main0()
```
But these cases don't 
```
from sympy import pi, Symbol, sin, cos

def main0():
    x: S = Symbol("x")
    print(sin(x).diff(x))
    print(sin(x).diff(x).diff(x))

main0()
```

### CODE REVIEW

**Comment by anutosh491** [comment]
Converting to draft as few corner cases need to be checked and the `symbolics_05.py` test file needs to be updated with these cases.

**Comment by anutosh491** [comment]
The has attribute can be added to the set of attributes once #2336 is merged. We can first merge this and then rebase that pr on top of this I think !

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -7220,6 +7241,32 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             ASR::expr_t* expr = ASR::down_cast<ASR::expr_t>(tmp);
             handle_builtin_attribute(expr, at->m_attr, loc, eles);
             return;
+        } else if (AST::is_a<AST::BinOp_t>(*at->m_value)) {
+            AST::BinOp_t* bop = AST::down_cast<AST::BinOp_t>(at->m_value);
+            std::set<std::string> symbolic_attributes = {
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/semantics/python_ast_to_asr.cpp:7246]: We can declare this as a Global variable, but it's fine for now.

File: integration_tests/symbolics_05.py [hunk 2]
```diff
@@ -21,4 +21,16 @@ def test_operations():
...
+    # test diff 2
+    c:S = sin(x)
+    d:S = cos(x)
+    assert(sin(Symbol("x")).diff(x) == d)
+    assert(sin(x).diff(Symbol("x")) == d)
+    assert(sin(x).diff(x) == d)
+    assert(sin(x).diff(x).diff(x) == S(-1)*c)
+    assert(sin(x).expand().diff(x).diff(x) == S(-1)*c)
+    assert((sin(x) + cos(x)).diff(x) == S(-1)*c + d)
+    assert((sin(x) + cos(x) + exp(x) + pi).diff(x).expand().diff(x) == exp(x) + S(-1)*c + S(-1)*d)
```
Comment (Thirumalai-Shaktivel) [inline integration_tests/symbolics_05.py:33]: Why don't we test this in the CI?

Comment (anutosh491) [reply integration_tests/symbolics_05.py:33]: I didn't catch you. Don't we execute `./run_tests.py -b=llvm_sym` and `./run_tests.py -b=c_sym` in `LPython CI / Run SymPy tests` ? I think so that checks this.

Comment (Thirumalai-Shaktivel) [reply integration_tests/symbolics_05.py:33]: Okay, cool.

**Comment by certik** [comment]
@czgdp1807, @Thirumalai-Shaktivel, @Shaikh-Ubaid any ideas how to best support this? We want to support custom SymPy methods, but do not want to pollute the frontend at the wrong places. I think this is similar to custom NumPy methods that we support (or will support).

**Comment by Thirumalai-Shaktivel** [comment]
As Anutosh said, the constant like pi will be treated as variable and it is not declared in current scope, the same goes for function call as well.
So for that, I think:
1. We can add a new node (let's say SymbolicItems or SymbolicObjects) and handle this in the Symbolic pass.
2. We can add more symbols like SymbolicFunction, SymbolicConstant, SymbolicBinOp.

**Review by certik — APPROVED** [review]
I don't know how to improve it. We need this feature, so let's merge this for now. We might need to improve it later, once we know how.
